lib/operations/viewsMaterialized.js

```diff
@@ -0,0 +1,96 @@
+import { template, quote, formatLines } from "../utils";
+
+const dataClause = data =>
+  data !== undefined ? ` WITH${data ? "" : " NO"} DATA` : "";
+
+export const drop = (viewName, { ifExists, cascade } = {}) =>
+  template`DROP MATERIALIZED VIEW${ifExists ? " IF EXISTS" : ""} "${viewName}"${
+    cascade ? " CASCADE" : ""
+  };`;
+
+export const create = (viewName, options, definition) => {
+  const {
+    ifNotExists,
+    columns = [],
+    tablespace,
+    storageParameters = {},
+    data
+  } = options;
+  const columnNames = quote(Array.isArray(columns) ? columns : [columns]).join(
+    ", "
+  );
+  const withOptions = Object.keys(storageParameters)
+    .map(
+      key =>
+        `${key}${
+          storageParameters[key] === true ? "" : ` = ${storageParameters[key]}`
+        }`
+    )
+    .join(", ");
+
+  return template`CREATE MATERIALIZED VIEW${
+    ifNotExists ? " IF NOT EXISTS" : ""
+  } "${viewName}"${columnNames ? `(${columnNames})` : ""}${
+    withOptions ? ` WITH (${withOptions})` : ""
+  }${tablespace ? `TABLESPACE ${tablespace}` : ""} AS ${definition}${dataClause(
+    data
+  )};`;
+};
+
+export const alter = (viewName, options) => {
+  const { cluster, extension, storageParameters = {} } = options;
+  const clauses = [];
+  if (cluster !== undefined) {
+    if (cluster) {
+      clauses.push(`CLUSTER ON "${cluster}"`);
+    } else {
+      clauses.push(`SET WITHOUT CLUSTER`);
+    }
+  }
+  if (extension) {
+    clauses.push(`DEPENDS ON EXTENSION "${extension}"`);
+  }
+  const withOptions = Object.keys(storageParameters)
+    .filter(key => storageParameters[key])
+    .map(
+      key =>
+        `${key}${
+          storageParameters[key] === true ? "" : ` = ${storageParameters[key]}`
+        }`
```

Comment:
... or this.

Reply:
https://github.com/prettier/prettier/issues/3368
I will look on it in some next PR. Should be "fixable" by storing these partials in constants...